sherpa/astro/io/crates_backend.py

```diff
@@ -1220,37 +1264,31 @@ def set_image_data(filename, data, header, ascii=False, clobber=False,
         raise IOErr('writenoimg')
 
     img.write(filename, clobber=True)
+    return None
 
 
 def set_table_data(filename, data, col_names, header=None,
-                   ascii=False, clobber=False, packup=False):
+                   ascii=False, clobber=False, packup=False) -> Optional[TABLECrate]:
 
     if not packup:
         check_clobber(filename, clobber)
 
     tbl = TABLECrate()
     hdr = {} if header is None else header
-    try:
-        for name in col_names:
-            _set_column(tbl, name, data[name])
-
-        for key, value in hdr.items():
-            if value is None:
-                continue
-            _set_key(tbl, key, value)
+    for name in col_names:
+        _set_column(tbl, name, data[name])
 
-    finally:
-        if packup:
-            return tbl
+    _update_header(tbl, hdr, skip_if_known=False)
 
-        if ascii and '[' not in filename and ']' not in filename:
-            filename += "[opt kernel=text/simple]"
+    if packup:
+        return tbl
 
-        tbl.write(filename, clobber=True)
+    write_dataset(tbl, filename, ascii=ascii)
+    return None
 
 
 def set_arf_data(filename, data, col_names, header=None,
-                 ascii=False, clobber=False, packup=False):
+                 ascii=False, clobber=False, packup=False) -> Optional[TABLECrate]:
     """Create an ARF"""
```

Comment:
Why is this optional? In case we pass in an empty dataset?

Reply:
The return value depends on the `packup` argument. So this could be done as an `overload` (ie when packup=False we return None and when True it returns the value), but a later PR changes this code to remove the `packup` option.